libglusterfs/src/inode.c

```diff
@@ -450,14 +450,16 @@ __inode_retire(inode_t *inode)
 static int
 __inode_get_xl_index(inode_t *inode, xlator_t *xlator)
 {
-    int set_idx = -1;
+    int set_idx = xlator->xl_id;
 
-    if ((inode->_ctx[xlator->xl_id].xl_key != NULL) &&
-        (inode->_ctx[xlator->xl_id].xl_key != xlator))
+    if (inode->_ctx[set_idx].xl_key == NULL) {
+        inode->_ctx[set_idx].xl_key = xlator;
         goto out;
```

Comment:
not needed?

Reply:
Setting xl_key when it is NULL is required. We can safely skip setting it in case it is already = xlator. Like https://github.com/gluster/glusterfs/pull/3208#issuecomment-1030775643

Reply:
The 'goto out' part is not needed.